src/spikeinterface/core/base.py

```diff
@@ -298,6 +298,23 @@ def to_dict(
         dump_dict: dict
             Serialized dictionary
         """
+        to_dict_kwargs = dict(
+            include_annotations=include_annotations,
+            include_properties=include_properties,
+            relative_to=relative_to,
+            folder_metadata=folder_metadata,
+        )
```

Comment:
I think that include annotation and properties should be maybe False when propagating recurssively in the lambda function. I think we do not need then.
Only the top level are needed.

The old behavior before refactoring was like this when to_dict() was done everywhere.

Reply:
Makes, sense, what about `folder_metadata`?

Reply:
Wait, on a second thought, would not that beat the purpose that @DradeAW wanted to use this as a provenance. It should be a faithful version all the way up.

Reply:
I don't mind having `include_annotations/properties` False by default, but I'd like the option to manually set it to true (yes I do want everything about the provenance, including all the annotations and properties).

Thanks everyone for all the hard good work!

Reply:
The question is more that we propagate ecussively properties or not.
I think we should not.
This is too heavy and useless.
And was not teh case before.

Reply:
OK. Check the latest comments on the other PR though:
https://github.com/SpikeInterface/spikeinterface/pull/1622#issuecomment-1562000001

The correct solution to pickle taking too long to load is now in :
https://github.com/SpikeInterface/spikeinterface/pull/1674

Reply:
Annotations aren’t heavy.
Also dumping to json doesn’t propagate properties. I thought the point of pickle was to contain all the information since it’s compressing everything